### PR TITLE
Propagate the KUBERNETES_SERVICE_XXX variables to API server.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Similarly to #1014, propagate the env vars to the API server.

## For PR author

- [x] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/apiserver/apiserver_controller.go [hunk 1]
```diff
@@ -36,28 +30,37 @@ import (
...
+	"github.com/tigera/operator/pkg/controller/utils"
+	"github.com/tigera/operator/pkg/render"
 )
 
 var log = logf.Log.WithName("controller_apiserver")
 
 // Add creates a new APIServer Controller and adds it to the Manager. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
-func Add(mgr manager.Manager, opts options.AddOptions) error {
+func Add(k8sServiceEp k8sapi.ServiceEndpoint, mgr manager.Manager, opts options.AddOptions) error {
```
Comment (tmjd) [inline pkg/controller/apiserver/apiserver_controller.go:47]: I don't think there is any need to pass the k8sServiceEp through here. We didn't in the core_controller.

I'm half tempted to suggest we don't even pass it into the render functions and just have the renders use it directly. But I think I like it as an input the renders, I think...

Comment (fasaxc) [reply pkg/controller/apiserver/apiserver_controller.go]: I like having it as an input to rendering; makes it easier to test.

File: pkg/controller/apiserver/apiserver_controller.go [hunk 2]
```diff
@@ -47,17 +49,18 @@ func Add(mgr manager.Manager, opts options.AddOptions) error {
 		// No need to start this controller.
 		return nil
 	}
-	return add(mgr, newReconciler(mgr, opts.DetectedProvider, opts.AmazonCRDExists))
+	return add(mgr, newReconciler(k8sapi.Endpoint, mgr, opts.DetectedProvider, opts.AmazonCRDExists))
```
Comment (tmjd) [inline pkg/controller/apiserver/apiserver_controller.go:52]: Also on core_controller we didn't pass the Endpoint here

File: pkg/controller/apiserver/apiserver_controller.go [hunk 3]
```diff
@@ -47,17 +49,18 @@ func Add(mgr manager.Manager, opts options.AddOptions) error {
...
 }
 
 // newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager, provider operatorv1.Provider, amazonCRDExists bool) *ReconcileAPIServer {
+func newReconciler(k8sServiceEp k8sapi.ServiceEndpoint, mgr manager.Manager, provider operatorv1.Provider, amazonCRDExists bool) *ReconcileAPIServer {
```
Comment (tmjd) [inline pkg/controller/apiserver/apiserver_controller.go:56]: Or here.

File: pkg/controller/apiserver/apiserver_controller.go [hunk 4]
```diff
@@ -47,17 +49,18 @@ func Add(mgr manager.Manager, opts options.AddOptions) error {
...
 	r := &ReconcileAPIServer{
 		client:          mgr.GetClient(),
 		scheme:          mgr.GetScheme(),
 		provider:        provider,
 		amazonCRDExists: amazonCRDExists,
 		status:          status.New(mgr.GetClient(), "apiserver"),
+		k8sServiceEp:    k8sServiceEp,
```
Comment (tmjd) [inline pkg/controller/apiserver/apiserver_controller.go:63]: Or store it in the Reconcile object either.

**Review by tmjd — COMMENTED** [review]
A few more comments similar to the last one. Do you think there is any benefit to pass and storing the Endpoint as part of the Reconcile struct?

**Review by tmjd — APPROVED** [review]
LGTM

**Pull request merged**